### ISSUE TITLE
Calendar Mode shows single all-day event as two days (Google Calendar)

### ISSUE THREAD

**Issue by nivolaco** [totaldebug/atomic-calendar-revive#544]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

When I have an all-day event set in my Google Calendar on Monday, the Calendar Mode will show a dot for Monday AND Tuesday. I believe this is due to the fact that Google Calendar sets all day events from 00:00 to 00:00 (overflows into next day). When I tried entering an event that is from 00:00 to 23:59, I only get a single dot on the corresponding day in Calendar Mode.  

### Expected behaviour

An single all-day event should only show one dot in Calendar Mode. Currently it shows two dots

### Card Version

v6.1.2

### Home Assistant Version

2021.11.2

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by marksie1988** [comment]
Interesting, I don't see this issue on my calendar, what timezone are you in? wonder if its a timezone conversion issue.

**Comment by nivolaco** [comment]
I'm on Mountain Standard Time

**Comment by marksie1988** [comment]
I am unable to reproduce this issue, I have changed my google calendar to MST and also updated the HA Timezone, I even changed my local timezone in case the browser was doing anything, However I still only see the events on a single day as expected. 

Timezones isn't something that I have had to deal with much of but I assume that this test should provide me the same result.

**Comment by jlambert121** [comment]
I have this issue as well on mountain time.  The calendar backing this is an apple calendar too (vs. google).

**Comment by Chrisgozd** [comment]
I see this using Apple calendars too.  I don't think it has to do with timezones.  

My full day event:
Start Time: Jan 1, 2022 0:00
End Time: Jan 2, 2022 0:00

These two lines are the problem: https://github.com/totaldebug/atomic-calendar-revive/blob/master/src/index.ts#L1237-L1238

I fixed it locally for myself, and removed all of the whitelist/blacklist/visibility checks so this isn't a copy and paste solution for everyone, but I replaced that if check I linked with:

```
var fullDay = event.startTime.hour() == 0 && event.endTime.hour() == 0 && event.startTime.isSameOrBefore(event.endTime, 'day');

if (fullDay ? dayjs(m.date).isBetween(event.startTime, event.endTime, 'day', '[)') : dayjs(m.date).isBetween(event.startTime, event.endTime, 'day', '[]'))
{
```

This works for:
1. all day events that are a single day
2. all day events that span multiple days
3. an event that spans multiple days but is not all day
